// @pnp/spfx-controls-react/lib/controls/filePicker/UploadFilePickerTab/UploadFilePickerTab.module.scss.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/* tslint:disable */
require('./UploadFilePickerTab.module.css');
var styles = {
    filePicker: 'filePicker_cee29819',
    tabsContainer: 'tabsContainer_cee29819',
    tabContainer: 'tabContainer_cee29819',
    nav: 'nav_cee29819',
    tabHeaderContainer: 'tabHeaderContainer_cee29819',
    breadcrumbNav: 'breadcrumbNav_cee29819',
    breadcrumbNavItem: 'breadcrumbNavItem_cee29819',
    tab: 'tab_cee29819',
    tabFiles: 'tabFiles_cee29819',
    scrollablePaneWrapper: 'scrollablePaneWrapper_cee29819',
    actionButtonsContainer: 'actionButtonsContainer_cee29819',
    actionButtons: 'actionButtons_cee29819',
    actionButton: 'actionButton_cee29819',
    tabHeader: 'tabHeader_cee29819',
    selectedTab: 'selectedTab_cee29819',
    header: 'header_cee29819',
    xlg: 'xlg_cee29819',
    xxlg: 'xxlg_cee29819',
    xxxlg: 'xxxlg_cee29819',
    xxxxlg: 'xxxxlg_cee29819',
    itemPickerTopBar: 'itemPickerTopBar_cee29819',
    commandBarNoChevron: 'commandBarNoChevron_cee29819',
    fileItem: 'fileItem_cee29819',
    folderItem: 'folderItem_cee29819',
    fileTypeIcon: 'fileTypeIcon_cee29819',
    fileTypeIconIcon: 'fileTypeIconIcon_cee29819',
    detailsListHeaderFileTypeIcon: 'detailsListHeaderFileTypeIcon_cee29819',
    iconColumnHeader: 'iconColumnHeader_cee29819',
    fileRow: 'fileRow_cee29819',
    folderRow: 'folderRow_cee29819',
    emptyFolder: 'emptyFolder_cee29819',
    emptyFolderImage: 'emptyFolderImage_cee29819',
    emptyFolderImageTag: 'emptyFolderImageTag_cee29819',
    emptyFolderTitle: 'emptyFolderTitle_cee29819',
    emptyFolderSubText: 'emptyFolderSubText_cee29819',
    emptyFolderMobile: 'emptyFolderMobile_cee29819',
    emptyFolderPc: 'emptyFolderPc_cee29819',
    emptyFolderIcon: 'emptyFolderIcon_cee29819',
    localTabSinglePreview: 'localTabSinglePreview_cee29819',
    localTabSinglePreviewImage: 'localTabSinglePreviewImage_cee29819',
    localTabLabel: 'localTabLabel_cee29819',
    localTabInput: 'localTabInput_cee29819',
};
exports.default = styles;
/* tslint:enable */
